proto/engine/v1/json_marshal_unmarshal.go

```diff
@@ -746,6 +896,224 @@ func (e *ExecutionPayloadDeneb) MarshalJSON() ([]byte, error) {
 	})
 }
 
+func (e *ExecutionPayloadElectra) MarshalJSON() ([]byte, error) {
+	transactions := make([]hexutil.Bytes, len(e.Transactions))
+	for i, tx := range e.Transactions {
+		transactions[i] = tx
+	}
+	baseFee := new(big.Int).SetBytes(bytesutil.ReverseByteOrder(e.BaseFeePerGas))
+	baseFeeHex := hexutil.EncodeBig(baseFee)
+	pHash := common.BytesToHash(e.ParentHash)
+	sRoot := common.BytesToHash(e.StateRoot)
+	recRoot := common.BytesToHash(e.ReceiptsRoot)
+	prevRan := common.BytesToHash(e.PrevRandao)
+	bHash := common.BytesToHash(e.BlockHash)
+	blockNum := hexutil.Uint64(e.BlockNumber)
+	gasLimit := hexutil.Uint64(e.GasLimit)
+	gasUsed := hexutil.Uint64(e.GasUsed)
+	timeStamp := hexutil.Uint64(e.Timestamp)
+	recipient := common.BytesToAddress(e.FeeRecipient)
+	logsBloom := hexutil.Bytes(e.LogsBloom)
+	withdrawals := e.Withdrawals
+	if withdrawals == nil {
+		withdrawals = make([]*Withdrawal, 0)
+	}
+	blobGasUsed := hexutil.Uint64(e.BlobGasUsed)
+	excessBlobGas := hexutil.Uint64(e.ExcessBlobGas)
+
+	return json.Marshal(ExecutionPayloadElectraJSON{
+		ParentHash:         &pHash,
+		FeeRecipient:       &recipient,
+		StateRoot:          &sRoot,
+		ReceiptsRoot:       &recRoot,
+		LogsBloom:          &logsBloom,
+		PrevRandao:         &prevRan,
+		BlockNumber:        &blockNum,
+		GasLimit:           &gasLimit,
+		GasUsed:            &gasUsed,
+		Timestamp:          &timeStamp,
+		ExtraData:          e.ExtraData,
+		BaseFeePerGas:      baseFeeHex,
+		BlockHash:          &bHash,
+		Transactions:       transactions,
+		Withdrawals:        withdrawals,
+		BlobGasUsed:        &blobGasUsed,
+		ExcessBlobGas:      &excessBlobGas,
+		WithdrawalRequests: WithdrawalRequestProtoToJson(e.WithdrawalRequests),
+		DepositRequests:    DepositRequestProtoToJson(e.DepositReceipts),
+	})
+}
+
+func (j *ExecutionPayloadElectraJSON) ElectraDepositReceipts() []*DepositReceipt {
+	rcpt := make([]*DepositReceipt, 0, len(j.DepositRequests))
```

Comment:
Is there a reason why we dont want to define the size `make([]*DepositReceipt, len(j.DepositRequests))`

Reply:
Not really, I can write it the other way if you prefer!

Reply:
changed it create the fixed size slices and assign to index